src/shotgunEventDaemon.py

```diff
@@ -507,10 +516,76 @@ def _mainLoop(self):
+                        pluginName = plugin.getName()
+
+                        if pluginName in skipPlugins:
+                            self.log.debug(
+                                "Not processing event #%d. Plugin %s is waiting for a gap to fill.",
+                                event["id"],
+                                pluginName,
+                            )
+                            continue
+
+                        if not plugin.isActive():
+                            self.log.debug(
+                                "Not processing event #%d. Plugin %s is inactive.",
+                                event["id"],
+                                pluginName,
+                            )
+                            continue
+
+                        nextEventId = plugin.getNextUnprocessedEventId()
+
+                        if nextEventId is None:
+                            msg = (
+                                "Plugin %s has no last event marker. Sending event #%d."
+                            )
+                            self.log.debug(msg, pluginName, event["id"])
+                            plugin.process(event)
+                        elif event["id"] < nextEventId:
+                            msg = "Last processed event for plugin %s is #%d, got #%d. Skipping."
+                            self.log.debug(
+                                msg, pluginName, plugin.lastEvent["id"], event["id"]
+                            )
+                        elif event["id"] == nextEventId:
+                            self.log.debug(
+                                "Sending expected event #%d to plugin %s.",
+                                event["id"],
+                                pluginName,
+                            )
+                            plugin.process(event)
+                        elif event["id"] > nextEventId:
+                            msg = "There is an event log entry gap between #%d (%s) and #%d (%s) for plugin %s."
+                            self.log.debug(
+                                msg,
+                                plugin.lastEvent["id"],
+                                plugin.lastEvent["created_at"],
+                                event["id"],
+                                event["created_at"],
+                                pluginName,
+                            )
+
+                            delta = event["created_at"] - plugin.lastEvent["created_at"]
+                            if delta >= datetime.timedelta(seconds=5 * 60):
+                                msg = "Gap delta is %s. Assuming events in the gap won't show up, sending event to plugin %s."
+                                self.log.debug(msg, pluginName)
+                                plugin.process(event)
+                            else:
+                                msg = "Gap delta is %s. Will wait until the gap fills up or the timeout is hit."
```

Comment:
would it be helpful to say what the timeout is in this statement, or is that overkill?

Reply:
Totally useful. Added.